SearchBar Typescript types are broken

<!--
Before filling out an issue:
1. A bug means that there is something broken or outside expectations in react-native-elements. If you only need help writing your own components, check out the [Slack channel](https://react-native-elements-slack.herokuapp.com/) FIRST.
2. If any section is left blank, there is a high possibility that the ticket will be closed. If you want your issue to get solved, you need to fill out ALL of the below fields. Otherwise, the ticket will likely be closed as not providing enough info.
-->

**Explain what you did (Required)**

I'm using SearchBar in typescript. I have a state variable to store the search text:

```
  const [search, setSearch] = React.useState<string>();
  return (
      <SearchBar
          onChangeText={setSearch}
        />
  )
```

**Expected behavior (Required)**

I expect this to be a valid usage according to Typescript.

**Describe the bug (Required)**

TS says onChangeText doesn't match, but the required property type is essentially disjoint (I think).

```
index.tsx(74,11): error TS2322: Type '(text: string) => void' is not assignable to type '((text: string) => void) & ((text: string) => void) & (() => any) & (() => any) & (() => any) & ((text: string) => void) & (() => any) & (() => any) & (() => any)'.
  Type '(text: string) => void' is not assignable to type '() => any'.
```

**Your Environment (Required):**

| software              | version |
| --------------------- | ------- |
| react-native-elements |     3.4.2    |
| react-native          |     0.64.2    |

--- a/src/SearchBar/SearchBar.tsx
+++ b/src/SearchBar/SearchBar.tsx
@@ -21,7 +21,7 @@ const SEARCHBAR_COMPONENTS = {
 export type SearchBarBaseProps = React.ComponentPropsWithRef<
   typeof TextInput
 > & {
-  platform: 'default' | 'ios' | 'android';
+  platform?: 'default' | 'ios' | 'android';
   containerStyle?: StyleProp<ViewStyle>;
   inputContainerStyle?: StyleProp<ViewStyle>;
   clearIcon?: IconNode;
@@ -38,10 +38,11 @@ export type SearchBarBaseProps = React.ComponentPropsWithRef<
   onCancel?(): void;
 };
 
-export type SearchBarProps = SearchBarBaseProps &
-  SearchBarDefaultProps &
-  SearchBarAndroidProps &
-  SearchBarIosProps;
+export type SearchBarProps =
+  | SearchBarBaseProps
+  | SearchBarDefaultProps
+  | SearchBarAndroidProps
+  | SearchBarIosProps;
 
 export class SearchBar extends React.Component<
   SearchBarProps & Partial<ThemeProps<SearchBarProps>>

--- a/src/SearchBar/SearchBar.tsx
+++ b/src/SearchBar/SearchBar.tsx
@@ -21,7 +21,7 @@ const SEARCHBAR_COMPONENTS = {
 export type SearchBarBaseProps = React.ComponentPropsWithRef<
   typeof TextInput
 > & {
-  platform?: 'default' | 'ios' | 'android';
+  platform?:  'default' | 'ios' | 'android';
   containerStyle?: StyleProp<ViewStyle>;
   inputContainerStyle?: StyleProp<ViewStyle>;
   clearIcon?: IconNode;

--- a/src/SearchBar/SearchBar.tsx
+++ b/src/SearchBar/SearchBar.tsx
@@ -21,7 +21,7 @@ const SEARCHBAR_COMPONENTS = {
 export type SearchBarBaseProps = React.ComponentPropsWithRef<
   typeof TextInput
 > & {
-  platform?:  'default' | 'ios' | 'android';
+  platform?: 'default' | 'ios' | 'android';
   containerStyle?: StyleProp<ViewStyle>;
   inputContainerStyle?: StyleProp<ViewStyle>;
   clearIcon?: IconNode;

